ci/define-example-runner.sh

```diff
@@ -49,19 +49,23 @@ run_example() {
   local example=$2
   shift 2
   local arguments=$*
-  local program_name=$(basename ${program_path})
+  local program_name="$(basename "${program_path}")"
 
-  if [ ! -x ${program_path} ]; then
+  if [ ! -x "${program_path}" ]; then
     echo "${COLOR_YELLOW}[  SKIPPED ]${COLOR_RESET}" \
         " ${program_name} is not compiled"
     return
   fi
   log="$(mktemp -t "run_example.XXXXXX")"
   echo    "${COLOR_GREEN}[ RUN      ]${COLOR_RESET}" \
       " ${program_name} ${example} running"
-  echo ${program_path} ${example} ${arguments} >"${log}"
+  # We want ${arguments} to expand as separate parameters.
```

Comment:
I would think we don't want `${arguments}` to expand as separate parameters, but rather that we want to pass along exactly the arguments we were given.  That is, something more along the lines of ...
```
local -a arguments=( "$@" )
${program_path} "${example}" "${arguments[@]}"
```